### PR TITLE
Format time to GMT for conditional headers

### LINKED ISSUE #20475: IfModified/IfUnmodifiedSince format can result in inconsistent behavior
### Bug Report

latest storage SDK

When testing source change detection, we want to see a failure when downloading the file since the LMT is newer than expected. When I send the IfUnmodifiedSince as PDT (my default machine time), the service succeeds with a 2XX (incorrectly). If I manually set it to GMT before passing it to the download API, the service fails with a 412 (expected behavior).

Fix : Convert all times sent to modified access conditions to GMT before sending. 

Note: The above is also a temporary workaround customers can use before passing their time to the SDK.

### PR DESCRIPTION
**Pull request by souravgupta-msft**
Fixes #20475 

- [x] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [x] Tests are included and/or updated for code changes.
- [x] Updates to [CHANGELOG.md][] are included.
- [x] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

File: sdk/storage/azblob/internal/generated/blob_client.go [hunk 1]
```diff
@@ -10,8 +10,12 @@ import (
 	"context"
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"time"
 )
 
+// used to convert times from UTC to GMT before sending across the wire
+var gmt = time.FixedZone("GMT", 0)
```
[the remark this answers is not in the document]
Comment (jhendrixMSFT) [reply sdk/storage/azblob/internal/generated/blob_client.go:17]: Thinking about this more, I don't think this is something we can simply fix in codegen.  OpenAPI specifies the time format but has no provisions to indicate if a value should be in UTC or not.  I will take a look at what other languages do.

[the remark this answers is not in the document]
Comment (jhendrixMSFT) [reply sdk/storage/azblob/internal/generated/blob_client.go:17]: Actually my solution might be wrong, please disregard for now.

**Comment by jhendrixMSFT** [comment]
@souravgupta-msft does this change address all the cases where this is an issue?

**Comment by souravgupta-msft** [comment]
> @souravgupta-msft does this change address all the cases where this is an issue?

Yes, I have added the transform for formatting these request headers:
- If-Modified-Since
- If-Unmodified-Since
- x-ms-source-if-modified-since
- x-ms-source-if-unmodified-since
- x-ms-immutability-policy-until-date

**Review by gapra-msft — APPROVED** [review]

**Review by siminsavani-msft — APPROVED** [review]

**Pull request merged**